core/src/main/scala/com/softwaremill/react/kafka_api_opt1/Consumer.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.softwaremill.react.kafka_api_opt1
+
+import scala.collection.immutable
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import akka.NotUsed
+import akka.stream.Graph
+import akka.stream.Inlet
+import akka.stream.Outlet
+import akka.stream.Shape
+import akka.stream.SourceShape
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.GraphDSL
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import org.apache.kafka.common.TopicPartition
+
+object Consumer {
+
+  // FIXME those should probably be real case classes (or value classes)?
+  type CommitMsg = Map[TopicPartition, OffsetAndMetadata]
+  type CommitConfirmation = Future[CommitMsg]
+
+  case class ConsumerShape[K, V](
+      commit: Inlet[CommitMsg],
+      messages: Outlet[ConsumerRecord[K, V]],
+      confirmation: Outlet[CommitConfirmation]
```

Comment:
Thanks for clarifying. Then I understand what you mean. I was only thinking of that as a syntactic difference (because the `commit` method would know the identifiers under the hood), but now I understand what you mean.